core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.kt

```diff
@@ -0,0 +1,155 @@
+
+import android.R.anim
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.MenuItem.OnActionExpandListener
+import androidx.appcompat.widget.SearchView
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import io.reactivex.disposables.CompositeDisposable
+import kotlinx.android.synthetic.main.activity_search.searchViewAnimator
+import kotlinx.android.synthetic.main.activity_search.search_list
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.id
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.coreActivityComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.viewModel
+import org.kiwix.kiwixmobile.core.extensions.setDistinctDisplayedChild
+import org.kiwix.kiwixmobile.core.search.adapter.SearchAdapter
+import org.kiwix.kiwixmobile.core.search.adapter.SearchDelegate.RecentSearchDelegate
+import org.kiwix.kiwixmobile.core.search.adapter.SearchDelegate.ZimSearchResultDelegate
+import org.kiwix.kiwixmobile.core.search.adapter.SearchListItem
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.ActivityResultReceived
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.ClickedSearchInText
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.CreatedWithIntent
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.ExitedSearch
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.OnItemClick
+import org.kiwix.kiwixmobile.core.search.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.search.viewmodel.SearchViewModel
+import org.kiwix.kiwixmobile.core.search.viewmodel.State
+import org.kiwix.kiwixmobile.core.search.viewmodel.State.Empty
+import org.kiwix.kiwixmobile.core.search.viewmodel.State.Initialising
+import org.kiwix.kiwixmobile.core.search.viewmodel.State.Results
+import org.kiwix.kiwixmobile.core.utils.SimpleTextListener
+import javax.inject.Inject
+
+class SearchActivity : BaseActivity() {
+
+  val activityComponent by lazy { coreActivityComponent }
+  @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+
+  private lateinit var searchView: SearchView
+  private lateinit var searchInTextMenuItem: MenuItem
+
+  private val searchViewModel by lazy { viewModel<SearchViewModel>(viewModelFactory) }
+  private val compositeDisposable = CompositeDisposable()
+  private val searchAdapter: SearchAdapter by lazy {
+    SearchAdapter(
+      RecentSearchDelegate(::onItemClick) {
+        searchViewModel.actions.offer(OnItemLongClick(it))
+      },
+      ZimSearchResultDelegate(::onItemClick)
+    )
+  }
+
+  override fun injection() {
```

Comment:
Can `injection` be moved to **BaseActivity**?

Reply:
No, we need to call the `inject(this specific class)` method because we are moving away from dagger-android and the `AndroidInjection.inject` method. The ticket to completely remove dagger-android is underway at the moment